nav2_map_server/README.md

```diff
@@ -62,83 +110,157 @@ occupied_thresh: 0.65
+- PointCloud PointCloud2 (sensor_msgs/msg/PointCloud2)
 
-## MapIO library
+## map_io_2d library
 
-`MapIO` library contains following API functions declared in `map_io.hpp` to work with
+`map_io_2d` library contains following API functions declared in `map_2d/map_io_2d.hpp` to work with
 OccupancyGrid maps:
 
 - loadMapYaml(): Load and parse the given YAML file
 - loadMapFromFile(): Load the image from map file and generate an OccupancyGrid
 - loadMapFromYaml(): Load the map YAML, image from map file and generate an OccupancyGrid
 - saveMapToFile(): Write OccupancyGrid map to file
 
+## map_io_3d library
+
+`map_io_3d` library contains following API functions declared in `map_3d/map_io_3D.hpp` to work with
+3D(PointCloud) maps:
+
+- loadMapYaml(): Load and parse the given YAML file
+- loadMapFromFile(): Load the pcd from map file and generate an PointCloud
+- loadMapFromYaml(): Load the map YAML, image from map file and generate an PointCloud
+- saveMapToFile(): Write PointCloud map to file
+
 ## Services
 
-As in ROS navigation, the `map_server` node provides a "map" service to get the map. See the nav_msgs/srv/GetMap.srv file for details.
+### 2D Map Saver and server
+As in ROS navigation, the `map_server_2d` node provides a "map" service to get the map. 
+See the nav_msgs/srv/GetMap.srv file for details.
 
-NEW in ROS2 Eloquent, `map_server` also now provides a "load_map" service and `map_saver` -
+NEW in ROS2 Eloquent, `map_server_2d` also now provides a "load_map" service and `map_saver_2d` -
 a "save_map" service. See nav2_msgs/srv/LoadMap.srv and nav2_msgs/srv/SaveMap.srv for details.
 
-For using these services `map_server`/`map_saver` should be launched as a continuously running
-`nav2::LifecycleNode` node. In addition to the CLI, `Map Saver` has a functionality of server
-handling incoming services. To run `Map Saver` in a server mode
-`nav2_map_server/launch/map_saver_server.launch.py` launch-file could be used.
+For using these services `map_server_2d`/`map_saver_2d` should be launched as a continuously running
+`nav2::LifecycleNode` node. In addition to the CLI, `MapSaver<nav_msgs::msg::OccupancyGrid>` has a functionality of 
+server
+handling incoming services. To run `MapSaver<nav_msgs::msg::OccupancyGrid>` in a server mode
+`nav2_map_server/launch/map_saver_server_2d.launch.py` launch-file could be used.
 
 Service usage examples:
 
-```
-$ ros2 service call /map_server/load_map nav2_msgs/srv/LoadMap "{map_url: /ros/maps/map.yaml}"
-$ ros2 service call /map_saver/save_map nav2_msgs/srv/SaveMap "{map_topic: map, map_url: my_map, image_format: pgm, map_mode: trinary, free_thresh: 0.25, occupied_thresh: 0.65}"
+```shell
+ros2 service call /map_server/load_map nav2_msgs/srv/LoadMap "{map_url: /ros/maps/map.yaml}"
+ros2 service call /map_saver/save_map nav2_msgs/srv/SaveMap "{map_topic: map, map_url: my_map, image_format: pgm, map_mode: trinary, free_thresh: 0.25, occupied_thresh: 0.65}"
 ```
 
+### 3D Map Saver and Server
```

Comment:
This should be re-arranged:
* All information about 3D services should be moved to the **Services** section
* You can make two sub-sections in **Services** section: "2D services" and "3D services". All information from this section is suitable for the last sub-section, I think.